From 96b4927b026b1cf56e9298f537d7ca517a5902b4 Mon Sep 17 00:00:00 2001
Message-Id: <96b4927b026b1cf56e9298f537d7ca517a5902b4.1428007496.git.feitong.yi@intel.com>
From: Namrta <namrta.salonie@intel.com>
Date: Fri, 20 Mar 2015 22:43:19 +0530
Subject: [PATCH 1/6] SQUASHME! [VPG]: drm/i915: Fix for potential NULL
 pointer dereference issue.

Check if the allocation was successful before usage.
Identified by static analysis tool.

SQUASHME! - This patch should be squashed into the following existing patch:
commit 32a0e4b7d75421f8a3c54f7831857a4ec0a4eecc
Change-Id: I1d6bb8e27a07ec7d32d532185e4ca5d22fe84034
Author: Peter Antoine <peter.antoine@intel.com>
Date:   Mon Dec 8 15:31:07 2014 +0000

    SQUASHME! [VPG]: drm/i915: potential deadlock fix

Issue: GMINL-7286
Change-Id: Iaea2d4bd49d51727d1430207b3e588ae9ee30de0
Signed-off-by: Namrta <namrta.salonie@intel.com>
---
 drivers/gpu/drm/i915/i915_cmd_parser.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_cmd_parser.c b/drivers/gpu/drm/i915/i915_cmd_parser.c
index 90d39e0..380092b 100644
--- a/drivers/gpu/drm/i915/i915_cmd_parser.c
+++ b/drivers/gpu/drm/i915/i915_cmd_parser.c
@@ -1284,7 +1284,7 @@ static int copy_user_space_cmds(struct drm_i915_private *dev_priv,
 	temp_cmd_table->count = args->cmd_count;
 	temp_cmd_table->table = drm_malloc_ab(sizeof(*temp_cmd_table->table),
 					 args->cmd_count);
-	if (!temp_cmd_table) {
+	if (!temp_cmd_table->table) {
 		drm_free_large(temp_cmd_table);
 		return -ENOMEM;
 	}
-- 
1.7.9.5

